hamilton/graph.py

```diff
@@ -981,7 +981,9 @@ def next_nodes_function(n: node.Node) -> List[node.Node]:
                 deps.append(dep)
             return deps
 
-        return self.directional_dfs_traverse(next_nodes_function, starting_nodes=final_vars)
+        return self.directional_dfs_traverse(
```

Comment:
I think this shouldn't change -- e.g. we should remove it from nodes if its just a runtime input in the level above. No reason for this to care about the strange 'inputs can be outputs too' requirement?

Reply:
I think the better way requires more changes and surgery. Will add a comment to say perhaps this shouldn't be here.